<a href="https://colab.research.google.com/github/gustavopierre/data_science_portfolio/blob/master/Gustavo_Moreira_Pierre.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>


[![author](https://github.com/gustavopierre/data_science_portfolio/blob/master/author-gustavopierre.png?raw=true)](https://www.linkedin.com/in/gustavo_pierre) [![](https://img.shields.io/badge/python-3.7+-blue.svg)](https://www.python.org/downloads/release/python-365/) [![contributions welcome](https://img.shields.io/badge/contributions-welcome-brightgreen.svg?style=flat)](https://github.com/gustavopierre/data_science_portfolio/issues)

<p align="center">
  <img src="https://github.com/gustavopierre/data_science_portfolio/blob/master/covid_cover.jpg?raw=true" alt="COVID-19" >
  <a href="https://www.pexels.com/pt-br/foto/biologia-brilhante-luminoso-iluminado-4031867/?utm_content=attributionCopyText&utm_medium=referral&utm_source=pexels">Foto de Edward Jenner no Pexels</a>
</p>

# **Análise dos Casos de COVID-19 no Brasil**
----
A COVID-19 é uma doença respiratória aguda causada pelo coronavirus SARS-CoV-2. Os coronavirus fazem parte de uma grande família de virus comuns em várias espécies de animais, tais como camelos, gado, gatos e morcegos. Em raros casos, podem infectar pessoas, como é o caso dos coronavirus MERS-CoV e SARS-CoV [[MinisteriodaSaude](https://coronavirus.saude.gov.br/sobre-a-doenca)].<br>
O primeiro caso de COVID-19 foi identificado em Wuhan, na província de Hubei, na República Popular da China, no dia 01 de dezembro de 2019. O novo virus mapeado não possui sintomas mais graves do que os provocados pelo coronavirus responsável pelo surto de SARG, que ocorreu em 2003, e do qual é muito semelhante genéticamente. No entanto, a COVID-19 possui um contágio maior [[WikipediaCOVID-19](https://pt.wikipedia.org/wiki/Pandemia_de_COVID-19) e [WikipediaSARS-CoV](https://pt.wikipedia.org/wiki/SARS-CoV)]. <br>

A Organização Mundial de Saúde (OMS) caracterizou a COVID-19 como pandemia em 11 de março de 2020, quando já havia mais de 118 mil casos em 114 países e 4,2 mil mortes [[OPASBrasil](https://www.paho.org/bra/index.php?option=com_content&view=article&id=6120:oms-afirma-que-covid-19-e-agora-caracterizada-como-pandemia&Itemid=812)].<br> 

No dia 26 de fevereiro deste ano foi anunciado o primeiro caso confirmado no Brasil de um homem de 61 anos que havia viajado para a Itália e deu entrada no Hospital Albert Einstein, em São Paulo [[Sanar](https://www.sanarmed.com/linha-do-tempo-do-coronavirus-no-brasil)]. Este paciente, segundo o Governo de São Paulo, se recuperou [[BBCNewsBrasil](https://www.bbc.com/portuguese/brasil-52334034)].<br>

No dia 06 de novembro de 2020 foram registrados no Brasil 5.633.747 casos, sendo que 5.104.195 se recuperaram e 162.038 morreram [[SUSAnalitico](https://susanalitico.saude.gov.br/extensions/covid-19_html/covid-19_html.html)].<br>

Neste primeiro projeto do [Bootcamp de Data Science Aplicada da Alura](https://www.alura.com.br/bootcamp/data-science-aplicada/matriculas-abertas), serão analisados os dados da COVID-19 no Brasil, passando pelas fases de obtenção dos dados, análise exploratória e exibição. O objetivo deste Projeto é fazer análises comparativas dos casos confirmados e mortes de COVID-19 por municípios e estados <del>e regiões</del> do Brasil, se há relação com o Índice de Desenvolvimento Humano Municipal (IDHM) e densidade demográfica (relação da população pela área). Serão exibidos também os números geograficamente distribuídos no mapa do Brasil. Ao final as conclusões das análises serão apresentadas bem como sugestões para trabalhos futuros.


## **Obtenção dos Dados**
O primeiro passo para atingir os objetivos deste Projeto é buscar os dados necessários. Sendo que o Projeto se baseará nos dados sobre COVID-19 das Secretarias Estaduais de Saúde consolidados pelo site [Brasil.io](https://brasil.io/datasets/). Este é um site mantido por Álvaro Justen e contribuidores e criado para disponibilizar dados públicos do poder público de maneira consolidada, em formato aberto e que permita consulta e download.
No entanto, para nossas análises serão necessários também dados do IDHM, população, das áreas físicas dos estados e municípios, das coordenadas geográfica dos municípios e as geometrias (dados geoespaciais) dos estados<del> e regiões</del>. Estes dois <del>três</del> últimos serão necessários para a exibição dos dados em um mapa.



### **1) Obtenção dos IDHM dos estados e municípios**
A busca do IDH dos estados brasileiros foi feito na internet. O Programa das Nações Unidas para o Desenvolvimento ([PNUD Brasil](https://www.br.undp.org/content/brazil/pt/home.html)), o Instituto de Pesquisa Econômica Aplicada ([Ipea](https://www.ipea.gov.br/portal/)) e a [Fundação João Pinheiro](http://novosite.fjp.mg.gov.br/) divulgaram em 2019 o IDHM dos estados brasileiros de 2012 a 2017. No site do [Atlas do
Desenvolvimento Humano
no Brasil](https://atlasbrasil.org.br/ranking) foi possível obter arquivo referente à 2017 no formato do Microsoft Excel (XLSX), disponibilizado com nome [idhm2017_estados.xlsx](https://github.com/gustavopierre/data_science_portfolio/blob/master/idhm2017_estados.xlsx?raw=true) no github.No entanto, no mesmo site, só foi possível baixar os dados de IDHM dos municípios referentes à 2010. Este arquivo foi disponibilizado como [idhm2010_municipios.xls](https://github.com/gustavopierre/data_science_portfolio/blob/master/idhm2010_municipios.xlsx?raw=true) no github.

### **2) Obtenção dos dados de população e área de cada estado**
Também na internet, mais precisamente no site do IBGE, foram obtidos os arquivos com os dados da [população](https://ftp.ibge.gov.br/Estimativas_de_Populacao/Estimativas_2020/) e das [áreas dos estados](https://www.ibge.gov.br/geociencias/organizacao-do-territorio/estrutura-territorial/15761-areas-dos-municipios.html?=&t=downloads). Estes arquivos foram salvos no github com os nomes [estimativa_pop_2020.xls](https://github.com/gustavopierre/data_science_portfolio/blob/master/estimativa_pop_2020.xls?raw=true) e [area_uf_2019.xls](https://github.com/gustavopierre/data_science_portfolio/blob/master/area_uf_2019.xls?raw=true), respectivamente.

### **3) Obtenção das coordenadas geográficas dos municípios<del> e dados de geometria de cada estado</del>**
As informações referentes às coordenadas geográficas dos municípios brasileiros foram encontradas no [forum do QGISBrasil](https://groups.google.com/g/qgisbrasil/c/0kJDo2KH0is) e salvo no github como [municipios_coord.csv](https://github.com/gustavopierre/data_science_portfolio/blob/master/municipios_coord.xls?raw=true). <del>O arquivo com a geometria dos estados foi obtido no [github do Gis Dataset Brasil](https://github.com/fititnt/gis-dataset-brasil/blob/master/uf/geojson/uf.json) e também foi disponibilizado no github como [uf.json](https://github.com/gustavopierre/data_science_portfolio/blob/master/uf.json?raw=true).</del> O arquivo com a geometria das regiões brsileiras foi obtido no site do [GISMAPS](https://gismaps.com.br/downloads/major-regions-of-brazil-geojson/) e disponibilizado no github como [regioes.geojson](https://github.com/gustavopierre/data_science_portfolio/blob/master/regioes.geojson?raw=true)

## **Configuração do Ambiente**
Para a carregamento, análise e exploração dos dados, usaremos Python. Para seu uso, serão necessárias as seguintes bibliotecas de funções que serão importadas para o ambiente:

* Pandas - uma biblioteca de uso fácil, que permite ao usuário explorar, manipular e visualizar os dados de um modo bastante eficiente;
* Numpy - uma biblioteca usada, principalmente, para realizar cálculos em vetores multidimensionais; e
* Matplotlib - uma biblioteca para visualização de dados;
* Seaborn - uma biblioteca de visualização baseada na Matplotlib.
* Folium - uma biblioteca que permite vizualização de dados manipulados em Python em um mapa interativo.
* <del>Geopandas - é um projeto open source que foi criado para tornar a manipulação das geometrias (dados geoespaciais) no Python uma tarefa mais simples; e</del>
* <del>Descartes - é uma biblioteca necessária para plotar polígonos em geopandas.</del><br>
Iniciamos o código instalando o Folium <del>Geopandas e Descartes</del> que, por padrão, não <del>vêm</del> vem instalado<del>s</del> no Google Colab.


In [105]:
print("Installing folium...")
!pip -q install folium
print('OK')
#print("Installing geopandas...")
#!pip -q install --upgrade geopandas
#print('OK')
#print("Installing descartes...")
#!pip -q install descartes
#print('OK')

Installing folium...
OK


In [106]:
#from google.colab import drive
#drive.mount('/content/drive')

In [107]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import folium
#import geopandas as gpd
sns.set() # notebook
pd.options.mode.chained_assignment = None;
pd.options.display.float_format = "{:.2f}".format

## **Análise Exploratória dos Dados**
Os dados, como dito antes, têm origem em cinco arquivos que possuem, respectivamente: 
* <del>a geometria dos estados e das regiões;</del>
* as coordenadas geográficas dos municípios; 
* o IDHM dos estados e dos municípios; 
* a população das regiões, estados e municípios; e  
* os dados da COVID-19.

Vamos analisá-los separadamente.

### **Dados com as Coordenadas Geográfica dos Municípios**
As coordenadas geográficas dos municípios estão em um arquivo no formato Microsoft Excel, com uma única pasta. Vamos carregar os dados para um dataframe, visualizar os dados (início e fim). 

In [108]:
# criando o dataframe latlong_municipios
latlong_municipios = pd.read_csv('https://github.com/gustavopierre/data_science_portfolio/raw/master/municipios_coord.csv', sep=';')
# exibindo os 5 primeiros registros
latlong_municipios.head()

GEOCODIGO_MUNICIPIO         NOME_MUNICIPIO  LONGITUDE  LATITUDE
0             1100015  ALTA FLORESTA D'OESTE     -62.00    -11.94
1             1100023              ARIQUEMES     -63.03     -9.91
2             1100031                 CABIXI     -60.54    -13.50
3             1100049                 CACOAL     -61.44    -11.43
4             1100056             CEREJEIRAS     -60.82    -13.20

In [109]:
# exibindo os 5 últimos registros
latlong_municipios.tail()

GEOCODIGO_MUNICIPIO  NOME_MUNICIPIO  LONGITUDE  LATITUDE
5561             5222054  VICENTINÓPOLIS     -49.81    -17.73
5562             5222203        VILA BOA     -47.06    -15.03
5563             5222302   VILA PROPÍCIO     -48.88    -15.45
5564             5300108        BRASÍLIA     -47.89    -15.79
5565                   
             NaN        nan       nan

Vamos ver a quantidade de registros existentes.

In [110]:
# quantidade de colunas e linhas no dataframe
print(f'Registros: {latlong_municipios.shape[0]} cidades\nColunas: {latlong_municipios.shape[1]}')

Registros: 5566 cidades
Colunas: 4


Vamos ver se existem dados faltantes e os tipos dos dados nos registros.

In [111]:
# exibindo informações de quantidade de não nulos e os tipos das colunas
latlong_municipios.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5566 entries, 0 to 5565
Data columns (total 4 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   GEOCODIGO_MUNICIPIO  5566 non-null   object 
 1   NOME_MUNICIPIO       5565 non-null   object 
 2   LONGITUDE            5565 non-null   float64
 3   LATITUDE             5565 non-null   float64
dtypes: float64(2), object(2)
memory usage: 174.1+ KB


Podemos observar que apenas os campos NOME_MUNICIPIO, LONGITUDE e LATITUDE possuem um valor nulo e, na visualização, podemos observar que é o último registro. Vamos apagar o último registro.

In [112]:
# apagando a linha que possui os valores nulos
latlong_municipios.dropna(inplace=True)
# exibindo os últimos cinco registros para verificar se o último registro foi mesmo apagada
latlong_municipios.tail()

GEOCODIGO_MUNICIPIO  NOME_MUNICIPIO  LONGITUDE  LATITUDE
5560             5222005      VIANÓPOLIS     -48.51    -16.74
5561             5222054  VICENTINÓPOLIS     -49.81    -17.73
5562             5222203        VILA BOA     -47.06    -15.03
5563             5222302   VILA PROPÍCIO     -48.88    -15.45
5564             5300108        BRASÍLIA     -47.89    -15.79

O Brasil possui 5.570 municípios, mas o arquivo só possui 5.565 municípios. Os 5 municípios criados foram:
* Pescaria Brava-SC - emancipado em 2003 - coordenadas geográficas: Latitude -28.4247 e Longitude -48.8956 - código do município: 4212650;
* Balneário Rincão-SC - emancipado em 2003 - coordenadas geográficas: Latitude -28.8344 e Longitude -49.2361 - código do município: 4220000;
* Mojuí dos Campos-PA - emancipado em 2012 - coordenadas geográficas: Latitude -2.6847 e Longitude -54.6431 - código do município: 1504752;
* Pinto Bandeira-RS - reemancipado em 2010 - coordenadas geográficas: Latitude -29.0978 e Longitude -51.4503 - código do município: 4314548; e
* Paraíso das Águas-MS - emancipado em 2003 - coordenadas geográficas: Latitude -19.0258 e Longitude -53.0103 - código do município: 5006275.

Fontes: [Cidade-Brasil](https://www.cidade-brasil.com.br/), [DBCity.com](https://pt.db-city.com/), [Wikipedia](https://pt.wikipedia.org/) e [IBGE-Código dos Municípios](https://www.ibge.gov.br/explica/codigos-dos-municipios.php).

Precisamos acrescentar os municípios neste dataframe e tornar mais amigável o nome das colunas.

In [113]:
# dados dos novos municipios a serem acrescentados
novos_municipios = [
                   ['4212650', 'PESCARIA BRAVA', -48.8956, -28.4247],
                   ['4220000', 'BALNEÁRIO RINCÃO', -49.2361, -28.8344],
                   ['1504752', 'MOJUÍ DOS CAMPOS', -54.6431, -2.6847],
                   ['4314548', 'PINTO BANDEIRA', -51.4503, -29.0978],
                   ['5006275', 'PARAÍSO DAS ÁGUAS', -53.0103, -19.0258]
]
# laço que insere cada novo município no dataframe
for novo_municipio in novos_municipios:
    latlong_municipios.loc[len(latlong_municipios)] = novo_municipio

# exibir a quantidade de municípios no dataframe
print(f'Registros: {latlong_municipios.shape[0]} cidades')
# alterando os nomes das colunas
latlong_municipios.columns=['cod_municipio', 'municipio', 'longitude', 'latitude']
# exibir os últimos municípios do dataframe
latlong_municipios.tail(7)

Registros: 5570 cidades


cod_municipio          municipio  longitude  latitude
5563       5222302      VILA PROPÍCIO     -48.88    -15.45
5564       5300108           BRASÍLIA     -47.89    -15.79
5565       4212650     PESCARIA BRAVA     -48.90    -28.42
5566       4220000   BALNEÁRIO RINCÃO     -49.24    -28.83
5567       1504752   MOJUÍ DOS CAMPOS     -54.64     -2.68
5568       4314548     PINTO BANDEIRA     -51.45    -29.10
5569       5006275  PARAÍSO DAS ÁGUAS     -53.01    -19.03

### **Dados de IDHM dos Estados e Municípios**

Como são arquivos separados, vamos carregar primeiro os dados do IDHM dos Estados. 

In [114]:
# criando o dataframe com idhm dos estados
idhm_estados = pd.read_excel('https://github.com/gustavopierre/data_science_portfolio/blob/master/idhm2017_estados.xlsx?raw=true')
# visualizando os 3 primeiros registros
idhm_estados.head(3)

Territorialidade  Posição IDHM  ...  Posição IDHM Longevidade  IDHM Longevidade
0  Distrito Federal             1  ...                         1              0.86
1         São Paulo             2  ...                         2              0.80
2    Santa Catarina             3  ...                         4              0.78

[3 rows x 9 columns]

Como só nos interessa o nome do estado e o IDHM, vamos excluir as demais colunas e renomear as que permanecem.

In [115]:
# selecionando as colunas
colunas_selecionadas = ['Territorialidade', 'IDHM']
# recriando o dataframe somente com as colunas selecionadas
idhm_estados = idhm_estados[colunas_selecionadas]
# renomeando as colunas
idhm_estados.columns = ['estado', 'idhm']
# exibindo informações do dataframe criado
idhm_estados.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27 entries, 0 to 26
Data columns (total 2 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   estado  27 non-null     object 
 1   idhm    27 non-null     float64
dtypes: float64(1), object(1)
memory usage: 560.0+ bytes


In [116]:
#exibindo somente os três primeiros registros do idhm dos estados
idhm_estados.head(3)

estado  idhm
0  Distrito Federal  0.85
1         São Paulo  0.83
2    Santa Catarina  0.81

Vamos agora criar o dataframe de IDHM dos municípios.

In [117]:
# criando o dataframe com idhm dos estados
idhm_municipios = pd.read_excel('https://github.com/gustavopierre/data_science_portfolio/blob/master/idhm2010_municipios.xlsx?raw=true')
# visualizando os 3 primeiros registros
idhm_municipios.head(3)

Territorialidade  ...  IDHM Longevidade
0  São Caetano do Sul (SP)  ...              0.89
1  Águas de São Pedro (SP)  ...              0.85
2       Florianópolis (SC)  ...              0.87

[3 rows x 9 columns]

In [118]:
# verificando a quantidade de regitros e colunas
idhm_municipios.shape

(5565, 9)

Como somente nos interessam as colunas de nome do municipio e IDHM, vamos excluir as demais e criar uma coluna de estado, que será preenchido com o estado que está em parênteses no nome do município). Como o dataframe das coordenadas geográficas possui nomes dos municípios em letras maiúsculas, vamos alterá-los para maiúsculas neste dataframe de IDHM.

In [119]:
# não é necessário selecionar as colunas porque a variável colunas_selecionadas definida anteriormente, já atende
# vamos recriar o dataframe somente com as colunas selecionadas
idhm_municipios = idhm_municipios[colunas_selecionadas]
# alterando os nomes das colunas
idhm_municipios.columns = ['municipio', 'idhm']
# vamos criar a coluna de sigla do estado e preenchê-la
idhm_municipios['uf'] = ''
# tornado as letras do nome em maiúsculas
idhm_municipios['municipio'] = idhm_municipios['municipio'].apply(lambda municipio: municipio.upper())
# exibindo os 4 primeiros registros
idhm_municipios.head(4)

municipio  idhm uf
0  SÃO CAETANO DO SUL (SP)  0.86   
1  ÁGUAS DE SÃO PEDRO (SP)  0.85   
2       FLORIANÓPOLIS (SC)  0.85   
3  BALNEÁRIO CAMBORIÚ (SC)  0.84

In [120]:
# tirando a sigla do estado do nome do municipio e colocando-na coluna uf
for index, row in idhm_municipios.iterrows():
    idhm_municipios.loc[index, 'municipio'] = row['municipio'][:-5]
    idhm_municipios.loc[index, 'uf'] = row['municipio'][-3:-1]
    
idhm_municipios.head(8)

municipio  idhm  uf
0  SÃO CAETANO DO SUL  0.86  SP
1  ÁGUAS DE SÃO PEDRO  0.85  SP
2       FLORIANÓPOLIS  0.85  SC
3  BALNEÁRIO CAMBORIÚ  0.84  SC
4             VITÓRIA  0.84  ES
5              SANTOS  0.84  SP
6             NITERÓI  0.84  RJ
7             JOAÇABA  0.83  SC

### **Dados de População dos Estados e Municípios**

O mesmo arquivo no formato Microsoft Excel tem pastas diferentes com informação de população estimada para 2020 de estado e municípios. Vamos ler uma pasta separadamente, criando o dataframe correspondente.

In [121]:
# criando dataframe do estado
estados_pop = pd.read_excel(
    'https://github.com/gustavopierre/data_science_portfolio/blob/master/estimativa_pop_2020.xls?raw=true', 
    sheet_name='BRASIL E UFs', 
    skiprows=1, 
    skipfooter=7
)
estados_pop

BRASIL E UNIDADES DA FEDERAÇÃO  Unnamed: 1 POPULAÇÃO ESTIMADA
0                          Brasil         nan          211755692
1                    Região Norte         nan           18672591
2                        Rondônia         nan            1796460
3                            Acre         nan             894470
4                        Amazonas         nan            4207714
5                         Roraima         nan             631181
6                            Pará         nan            8690745
7                           Amapá         nan             861773
8                       Tocantins         nan            1590248
9                 Região Nordeste         nan           57374243
10                       Maranhão         nan            7114598
11                          Piauí         nan       3.281.480(1)
12                          Ceará         nan       9.187.103(1)
13            Rio Grande do Norte         nan            3534165
14                        Paraíba         nan            4039277
15                     Pernambuco         nan       9.616.621(2)
16                        Alagoas         nan       3.351.543(2)
17                        Sergipe         nan       2.318.822(3)
18                          Bahia         nan      14.930.634(3)
19                 Região Sudeste         nan           89012240
20                   Minas Gerais         nan           21292666
21                 Espírito Santo         nan            4064052
22                 Rio de Janeiro         nan           17366189
23                      São Paulo         nan           46289333
24                     Região Sul         nan           30192315
25                         Paraná         nan           11516840
26                 Santa Catarina         nan            7252502
27              Rio Grande do Sul         nan           11422973
28            Região Centro-Oeste         nan           16504303
29             Mato Grosso do Sul         nan            2809394
30                    Mato Grosso         nan            3526220
31                          Goiás         nan       7.113.540(4)
32               Distrito Federal         nan       3.055.149(4)

Podemos observar que existe uma coluna que não interessa e que todos seus valores são nulos. Nós iremos exclui-las. E podemos observar também que esses dados incluem a população das regiões. Vamos excluí-las, pois as regiões não serão escopo deste trabalho.

In [122]:
# excluindo a coluna de valores nulos
estados_pop.dropna(axis=1, inplace=True)
estados_pop.columns=['estado','populacao']
estados_pop.head()

estado  populacao
0        Brasil  211755692
1  Região Norte   18672591
2      Rondônia    1796460
3          Acre     894470
4      Amazonas    4207714

In [123]:
# removendo os registros das regiões
regioes = estados_pop.loc[(estados_pop['estado'].isin(['Brasil','Região Norte','Região Sul', 'Região Centro-Oeste', 'Região Nordeste', 'Região Sudeste']))]

estados_pop = estados_pop.drop(regioes.index)
estados_pop = estados_pop.reset_index(drop=True)
estados_pop

estado      populacao
0              Rondônia        1796460
1                  Acre         894470
2              Amazonas        4207714
3               Roraima         631181
4                  Pará        8690745
5                 Amapá         861773
6             Tocantins        1590248
7              Maranhão        7114598
8                 Piauí   3.281.480(1)
9                 Ceará   9.187.103(1)
10  Rio Grande do Norte        3534165
11              Paraíba        4039277
12           Pernambuco   9.616.621(2)
13              Alagoas   3.351.543(2)
14              Sergipe   2.318.822(3)
15                Bahia  14.930.634(3)
16         Minas Gerais       21292666
17       Espírito Santo        4064052
18       Rio de Janeiro       17366189
19            São Paulo       46289333
20               Paraná       11516840
21       Santa Catarina        7252502
22    Rio Grande do Sul       11422973
23   Mato Grosso do Sul        2809394
24          Mato Grosso        3526220
25                Goiás   7.113.540(4)
26     Distrito Federal   3.055.149(4)

Alguns valores de população tem um número entre parênteses. Estes números são referências a observações que são escritas no rodapé do arquivo, descartadas na leitura e irrelevantes para este trabalho. 

Estes números entre parênteses fez com que esses valores fossem lidos como texto (strings - aqui chamados 'objects') e não números, inclusive trazendo os pontos de separação de milhares e milhões. Vamos excluir este números entre parênteses e os pontos. Depois tranformamos os valores dessa coluna em números inteiros.

In [124]:
# verificando os tipos dos dados do dataframe e se há valores nulos
estados_pop.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 27 entries, 0 to 26
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   estado     27 non-null     object
 1   populacao  27 non-null     object
dtypes: object(2)
memory usage: 560.0+ bytes


In [125]:
# excluindo os parênteses, os números dentro deles e os pontos de cada registro que possui uma abertura de parêntese
for index, row in estados_pop.iterrows():
    if '(' in str(row['populacao']):
        estados_pop.loc[index,'populacao'] = ''.join(row['populacao'].split('(')[0].split('.'))

In [126]:
# exibindo os 3 últimos registros
estados_pop.tail(3)

estado populacao
24       Mato Grosso   3526220
25             Goiás   7113540
26  Distrito Federal   3055149

In [127]:
# transformando os dados do campo em número inteiro
estados_pop['populacao'] = estados_pop['populacao'].astype('int64')
estados_pop.dtypes


estado       object
populacao     int64
dtype: object

Vamos criar o dataframe das populações dos municípios que se encontra no mesmo arquivo Micrsoft Excel, só que na pasta 'Municípios'.

In [128]:
# criando o dataframe de população dos municípios
municipios_pop = pd.read_excel(
    'https://github.com/gustavopierre/data_science_portfolio/blob/master/estimativa_pop_2020.xls?raw=true', 
    sheet_name='Municípios', 
    skiprows=1, 
    skipfooter=16,
)
municipios_pop

UF  COD. UF  COD. MUNIC      NOME DO MUNICÍPIO POPULAÇÃO ESTIMADA
0     RO       11          15  Alta Floresta D'Oeste              22728
1     RO       11          23              Ariquemes             109523
2     RO       11          31                 Cabixi               5188
3     RO       11          49                 Cacoal              85893
4     RO       11          56             Cerejeiras              16204
...   ..      ...         ...                    ...                ...
5565  GO       52       22005             Vianópolis              13977
5566  GO       52       22054         Vicentinópolis               8873
5567  GO       52       22203               Vila Boa               6312
5568  GO       52       22302          Vila Propício               5882
5569  DF       53         108               Brasília            3055149

[5570 rows x 5 columns]

Podemos observar que temos os 5570 municípios do Brasil. 

Vamos verificar os tipos dos dados.

In [129]:
# verificando os tipos dos dados
municipios_pop.dtypes

UF                    object
COD. UF                int64
COD. MUNIC             int64
NOME DO MUNICÍPIO     object
POPULAÇÃO ESTIMADA    object
dtype: object

No momento da leitura, a coluna do código do estado e do código do municipio foram convertidos para inteiro. Vamos reler o arquivo colocando parâmetros que façam ele ler como texto esses dois campos e, em seguida, vamos alterar os nomes dos campos.

In [130]:
# criando o dataframe de população dos municípios
municipios_pop = pd.read_excel(
    'https://github.com/gustavopierre/data_science_portfolio/blob/master/estimativa_pop_2020.xls?raw=true', 
    sheet_name='Municípios', 
    skiprows=1, 
    skipfooter=16,
    dtype={'COD. UF': 'object', 'COD. MUNIC': 'object'}
)
# exibindo os registros
municipios_pop

UF COD. UF COD. MUNIC      NOME DO MUNICÍPIO POPULAÇÃO ESTIMADA
0     RO      11      00015  Alta Floresta D'Oeste              22728
1     RO      11      00023              Ariquemes             109523
2     RO      11      00031                 Cabixi               5188
3     RO      11      00049                 Cacoal              85893
4     RO      11      00056             Cerejeiras              16204
...   ..     ...        ...                    ...                ...
5565  GO      52      22005             Vianópolis              13977
5566  GO      52      22054         Vicentinópolis               8873
5567  GO      52      22203               Vila Boa               6312
5568  GO      52      22302          Vila Propício               5882
5569  DF      53      00108               Brasília            3055149

[5570 rows x 5 columns]

In [131]:
# alterando os nomes das colunas
municipios_pop.columns=['uf','cod_uf', 'cod_munic', 'municipio','populacao']

In [132]:
# verificando os tipos das colunas
municipios_pop.dtypes

uf           object
cod_uf       object
cod_munic    object
municipio    object
populacao    object
dtype: object

O campo da populacao está como texto. Vamos procurar se ocorre o mesmo que o anterior, com números entre parênteses.

In [133]:
# verificando se sem valores de populacao fechando parênteses
municipios_pop.populacao.str.contains(')', regex=False).sum()

12

Vamos, então, tirar os parênteses, os valores entre os parênteses e os pontos dos valores de população que os possui, como no caso da populacção dos estados.

In [134]:
# excluindo os parênteses, os números dentro deles e os pontos de cada registro que possui uma abertura de parêntese
for index, row in municipios_pop.iterrows():
    if '(' in str(row['populacao']):
        municipios_pop.loc[index,'populacao'] = ''.join(row['populacao'].split('(')[0].split('.'))

municipios_pop

uf cod_uf cod_munic              municipio populacao
0     RO     11     00015  Alta Floresta D'Oeste     22728
1     RO     11     00023              Ariquemes    109523
2     RO     11     00031                 Cabixi      5188
3     RO     11     00049                 Cacoal     85893
4     RO     11     00056             Cerejeiras     16204
...   ..    ...       ...                    ...       ...
5565  GO     52     22005             Vianópolis     13977
5566  GO     52     22054         Vicentinópolis      8873
5567  GO     52     22203               Vila Boa      6312
5568  GO     52     22302          Vila Propício      5882
5569  DF     53     00108               Brasília   3055149

[5570 rows x 5 columns]

Vamos transformar o campo populacao em inteiro e vamos tranformar os nomes dos municípios para maiúsculas.

In [135]:
# transformando os dados do campo em número inteiro
municipios_pop['populacao'] = municipios_pop['populacao'].astype('int64')

municipios_pop.dtypes

uf           object
cod_uf       object
cod_munic    object
municipio    object
populacao     int64
dtype: object

In [136]:
# tornado as letras do nome do município em maiúsculas
municipios_pop['municipio'] = municipios_pop['municipio'].apply(lambda municipio: municipio.upper())

# exibindo os 3 primeiros registros
municipios_pop.head(3)

uf cod_uf cod_munic              municipio  populacao
0  RO     11     00015  ALTA FLORESTA D'OESTE      22728
1  RO     11     00023              ARIQUEMES     109523
2  RO     11     00031                 CABIXI       5188

### **Dados de Área dos Estados e Municípios**

O mesmo arquivo no formato Microsoft Excel tem pastas diferentes com informação de área de estado e municípios. Vamos ler cada pasta separadamente, criando o dataframe correspondente.

In [137]:
# criando dataframe de áreas dos estados
estados_areas = pd.read_excel(
    'https://github.com/gustavopierre/data_science_portfolio/blob/master/area_uf_2019.xls?raw=true', 
    sheet_name='AR_BR_UF_2019',
)
estados_areas

ID  CD_GCUF                NM_UF NM_UF_SIGLA  AR_MUN_2019
0   1.00    11.00             Rondônia          RO    237765.24
1   2.00    12.00                 Acre          AC    164123.96
2   3.00    13.00             Amazonas          AM   1559167.89
3   4.00    14.00              Roraima          RR    223644.53
4   5.00    15.00                 Pará          PA   1245870.80
5   6.00    16.00                Amapá          AP    142470.76
6   7.00    17.00            Tocantins          TO    277466.76
7   8.00    21.00             Maranhão          MA    329642.18
8   9.00    22.00                Piauí          PI    251756.52
9  10.00    23.00                Ceará          CE    148894.44
10 11.00    24.00  Rio Grande do Norte          RN     52809.60
11 12.00    25.00              Paraíba          PB     56467.24
12 13.00    26.00           Pernambuco          PE     98067.88
13 14.00    27.00              Alagoas          AL     27843.29
14 15.00    28.00              Sergipe          SE     21925.42
15 16.00    29.00                Bahia          BA    564760.43
16 17.00    31.00         Minas Gerais          MG    586521.12
17 18.00    32.00       Espirito Santo          ES     46074.45
18 19.00    33.00       Rio de Janeiro          RJ     43750.43
19 20.00    35.00            São Paulo          SP    248219.48
20 21.00    41.00               Paraná          PR    199298.98
21 22.00    42.00       Santa Catarina          SC     95730.68
22 23.00    43.00    Rio Grande do Sul          RS    281707.16
23 24.00    50.00   Mato Grosso do Sul          MS    357145.53
24 25.00    51.00          Mato Grosso          MT    903207.02
25 26.00    52.00                Goiás          GO    340203.33
26 27.00    53.00     Distrito Federal          DF      5760.78
27   nan      nan                  NaN         NaN          nan
28   nan      nan                  NaN         NaN          nan
29   nan      nan                  NaN         NaN   8510295.91

O código IBGE da Unidade da Federação (CD_GCUF) está como número, deveria ser texto, pois fica mais fácil de concatenar com o código do município, se necessário. Quanto ao final, podemos observar 3 linhas finais que podemos excluir. Vamos eliminar a coluna ID, que não tem utilidade neste trabalho e renomear as colunas. Começamos criando novamente o dataframe.

In [138]:
# criando dataframe de áreas dos estados,sem ler as três últimas linhas e definindo que a coluna CD_GCUF deve ser lida como texto (object)
estados_areas = pd.read_excel(
    'https://github.com/gustavopierre/data_science_portfolio/blob/master/area_uf_2019.xls?raw=true', 
    sheet_name='AR_BR_UF_2019',
    dtype={'CD_GCUF': 'object'},
    skipfooter=3
)
# apagando a coluna ID
del estados_areas['ID']
# renomeando as colunas
estados_areas.columns = ['cod_uf', 'estado', 'sigla', 'area']
# exibindo o dataframe
estados_areas

cod_uf               estado sigla       area
0      11             Rondônia    RO  237765.24
1      12                 Acre    AC  164123.96
2      13             Amazonas    AM 1559167.89
3      14              Roraima    RR  223644.53
4      15                 Pará    PA 1245870.80
5      16                Amapá    AP  142470.76
6      17            Tocantins    TO  277466.76
7      21             Maranhão    MA  329642.18
8      22                Piauí    PI  251756.52
9      23                Ceará    CE  148894.44
10     24  Rio Grande do Norte    RN   52809.60
11     25              Paraíba    PB   56467.24
12     26           Pernambuco    PE   98067.88
13     27              Alagoas    AL   27843.29
14     28              Sergipe    SE   21925.42
15     29                Bahia    BA  564760.43
16     31         Minas Gerais    MG  586521.12
17     32       Espirito Santo    ES   46074.45
18     33       Rio de Janeiro    RJ   43750.43
19     35            São Paulo    SP  248219.48
20     41               Paraná    PR  199298.98
21     42       Santa Catarina    SC   95730.68
22     43    Rio Grande do Sul    RS  281707.16
23     50   Mato Grosso do Sul    MS  357145.53
24     51          Mato Grosso    MT  903207.02
25     52                Goiás    GO  340203.33
26     53     Distrito Federal    DF    5760.78

Vamos criar o dataframe a partir da pasta de áreas do mesmo arquivo anterior.

In [139]:
# criando dataframe de áreas dos municípios
municipios_areas = pd.read_excel(
    'https://github.com/gustavopierre/data_science_portfolio/blob/master/area_uf_2019.xls?raw=true', 
    sheet_name='AR_BR_MUN_2019',
)
municipios_areas

ID  CD_GCUF  ...            NM_MUN_2019 AR_MUN_2019
0       1.00    11.00  ...  ALTA FLORESTA D'OESTE     7067.02
1       2.00    11.00  ...              ARIQUEMES     4426.57
2       3.00    11.00  ...                 CABIXI     1314.35
3       4.00    11.00  ...                 CACOAL     3792.89
4       5.00    11.00  ...             CEREJEIRAS     2783.30
...      ...      ...  ...                    ...         ...
5570 5571.00    52.00  ...          VILA PROPÍCIO     2181.58
5571 5572.00    53.00  ...               BRASÍLIA     5760.78
5572     nan      nan  ...                    NaN         nan
5573     nan      nan  ...                    NaN         nan
5574     nan      nan  ...                    NaN  8510295.91

[5575 rows x 7 columns]

Vamos fazer algo bem semelhante ao dataframe anterior: ler somente os dados sem ler os três últimos registros, informar que os dados do código da unidade da federação (CD_GCUF) e o código do município (CD_GCMUN) são textos, apagar a coluna ID e renomear as colunas.

In [140]:
# criando dataframe de áreas dos municípios, sem ler as três últimas linhas e 
# definindo que a coluna CD_GCUF e CD_GCMUN devem ser lidas como texto (object)
municipios_areas = pd.read_excel(
    'https://github.com/gustavopierre/data_science_portfolio/blob/master/area_uf_2019.xls?raw=true', 
    sheet_name='AR_BR_MUN_2019',
    dtype={'CD_GCUF': 'object', 'CD_GCMUN': 'object'},
    skipfooter=3
)
# apagando a coluna ID
del municipios_areas['ID']
# renomeando as colunas
municipios_areas.columns = ['cod_uf', 'estado', 'sigla', 'cod_munic', 'municipio', 'area']
# exibindo o dataframe
municipios_areas

cod_uf            estado sigla cod_munic              municipio    area
0        11          Rondônia    RO   1100015  ALTA FLORESTA D'OESTE 7067.02
1        11          Rondônia    RO   1100023              ARIQUEMES 4426.57
2        11          Rondônia    RO   1100031                 CABIXI 1314.35
3        11          Rondônia    RO   1100049                 CACOAL 3792.89
4        11          Rondônia    RO   1100056             CEREJEIRAS 2783.30
...     ...               ...   ...       ...                    ...     ...
5567     52             Goiás    GO   5222005             VIANÓPOLIS  954.28
5568     52             Goiás    GO   5222054         VICENTINÓPOLIS  737.25
5569     52             Goiás    GO   5222203               VILA BOA 1060.17
5570     52             Goiás    GO   5222302          VILA PROPÍCIO 2181.58
5571     53  Distrito Federal    DF   5300108               BRASÍLIA 5760.78

[5572 rows x 6 columns]

Podemos ver que temos 5572 municípios, sendo dois a mais do que a quantidade existente de municípios no Brasil. Vamos verificar se existe algum registro com valor nulo. 

In [141]:
# Verificando se tem algum valor nulo
municipios_areas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5572 entries, 0 to 5571
Data columns (total 6 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   cod_uf     5572 non-null   object 
 1   estado     5572 non-null   object 
 2   sigla      5572 non-null   object 
 3   cod_munic  5572 non-null   object 
 4   municipio  5572 non-null   object 
 5   area       5572 non-null   float64
dtypes: float64(1), object(5)
memory usage: 261.3+ KB


Não existem valores negativos. Essa informação é importante no momento que juntarmos dataframes.

### **Dados da COVID-19**

Não foi possível, de forma simples, armazenar os dados da COVID-19 no github, portanto, ele foi armazenado no dropbox para sua leitura direta. Vamos ler o arquivo e criar o dataframe com os dados.

In [142]:
# criando o dataframe a partir do arquivo de dados armazenado no dropbox
dados = pd.read_csv('https://www.dropbox.com/s/o0i2mjb3zdgoif1/caso.csv?dl=1')
# exibindo os primeiros cinco registros
dados.head()

date state  ... confirmed_per_100k_inhabitants death_rate
0  2020-11-11    AP  ...                        6241.78       0.01
1  2020-11-10    AP  ...                        6216.72       0.01
2  2020-11-09    AP  ...                        6188.40       0.01
3  2020-11-08    AP  ...                        6151.27       0.01
4  2020-11-03    AP  ...                        6130.62       0.01

[5 rows x 13 columns]

In [143]:
# exibindo a quantidade de registros e colunas
dados.shape

(957146, 13)

Vamos ler novamente o arquivo, de forma que o código da cidade venha como texto.

In [144]:
# criando o dataframe a partir do arquivo de dados armazenado no dropbox
dados = pd.read_csv('https://www.dropbox.com/s/o0i2mjb3zdgoif1/caso.csv?dl=1',
                    dtype={'city_ibge_code': 'object'})
# exibindo os primeiros cinco registros
dados.head()

date state  ... confirmed_per_100k_inhabitants death_rate
0  2020-11-11    AP  ...                        6241.78       0.01
1  2020-11-10    AP  ...                        6216.72       0.01
2  2020-11-09    AP  ...                        6188.40       0.01
3  2020-11-08    AP  ...                        6151.27       0.01
4  2020-11-03    AP  ...                        6130.62       0.01

[5 rows x 13 columns]

Com base nas informações sobre o arquivo no [site da documentação do arquivo](https://github.com/turicas/covid19-br/blob/master/api.md#casos), as colunas têm os seguintes significados:
* **date**: data de coleta dos dados;
* **state**: sigla da unidade federativa;
* **city**: nome do município (pode estar em branco quando o registro é referente ao estado, pode ser preenchido com Importados/Indefinidos também);
* **place_type**: tipo de local que esse registro descreve, pode ser city ou state;
* **order_for_place**: número que identifica a ordem do registro para este local. O registro referente ao primeiro boletim em que esse local aparecer será contabilizado como 1 e os demais boletins incrementarão esse valor;
* **is_last**: campo que diz se esse registro é o mais novo para esse local
* **city_ibge_code**: código IBGE do município;
* **confirmed**: número de casos confirmados;
* **deaths**: número de mortes;
* **estimated_population**: população estimada para esse município/estado em 2020, segundo o IBGE;
* **estimated_population_2019**: população estimada para esse município/estado em 2019, segundo o IBGE;
* **confirmed_per_100k_inhabitants**: número de casos confirmados por 100.000 habitantes; e
* **death_rate**: taxa de mortalidade (mortes / confirmados).



Vamos eliminar as seguintes colunas:
* a data de coleta, pois os valores sendo acumulativos, só nos interessa, para este trabalho, o último valor que é identificado pela coluna **is_last** igual a verdadeiro (True);
* a coluna de ordem do registro, pois não tem utilidade para este trabalho;
* de população estimada em 2019 - porque não tem utilidade para este trabalho; e
* de casos  confirmados por cem mil habitantes, pois essa quantidade de casos serão contabilizados em relação mil habitantes.

In [145]:
# apagando as colunas
dados = dados.drop(columns=['date','order_for_place', 'estimated_population_2019', 'confirmed_per_100k_inhabitants'])

In [146]:
# exibindo os 5 primeiros registros
dados.head()

state city place_type  ...  estimated_population  city_ibge_code  death_rate
0    AP  NaN      state  ...             861773.00              16        0.01
1    AP  NaN      state  ...             861773.00              16        0.01
2    AP  NaN      state  ...             861773.00              16        0.01
3    AP  NaN      state  ...             861773.00              16        0.01
4    AP  NaN      state  ...             861773.00              16        0.01

[5 rows x 9 columns]

Vamos apagar os dados que não são os últimos.

In [147]:
# redefinindo o dataframe para conter somente os últimos registros
dados = dados.query("is_last == True")
dados.head()

state            city  ... city_ibge_code  death_rate
0      AP             NaN  ...             16        0.01
232    AP           Amapá  ...        1600105        0.01
419    AP        Calçoene  ...        1600204        0.00
601    AP          Cutias  ...        1600212        0.00
779    AP  Ferreira Gomes  ...        1600238        0.01

[5 rows x 9 columns]

Vamos ver os dados nulos existentes.

In [148]:
# visualizando a quantidade de dados nulos
dados.isnull().sum()

state                    0
city                    27
place_type               0
confirmed                0
deaths                   0
is_last                  0
estimated_population    19
city_ibge_code          19
death_rate               0
dtype: int64

Os registros referentes aos estados têm cidades como nulos e usa o código do municpio como código da unidade da federação. Isso explica ter  27 registros com o campo cidade como nulos e porque não ter 27 registros  nulos no campo código do município. No entanto, temos 19 registros nulos na população e na coluna código do município. Vamos tentar descobrir se são registros úteis para este trabalho.

In [149]:
# listando os registros que têm a população nula
dados.loc[dados.estimated_population.isnull()]

state                    city  ... city_ibge_code  death_rate
8214      DF  Importados/Indefinidos  ...            NaN        0.01
24352     CE  Importados/Indefinidos  ...            NaN        0.00
51429     ES  Importados/Indefinidos  ...            NaN        0.02
87943     BA  Importados/Indefinidos  ...            NaN        0.01
207758    GO  Importados/Indefinidos  ...            NaN        0.00
236004    MT  Importados/Indefinidos  ...            NaN        0.13
254719    PA  Importados/Indefinidos  ...            NaN        0.00
288366    PB  Importados/Indefinidos  ...            NaN        0.00
365952    MG  Importados/Indefinidos  ...            NaN        0.03
451284    AL  Importados/Indefinidos  ...            NaN        1.00
479485    PE  Importados/Indefinidos  ...            NaN        0.00
564148    PR  Importados/Indefinidos  ...            NaN        0.02
606916    RO  Importados/Indefinidos  ...            NaN        0.00
630250    RJ  Importados/Indefinidos  ...            NaN        0.00
650898    RN  Importados/Indefinidos  ...            NaN        0.03
669314    RR  Importados/Indefinidos  ...            NaN        0.00
704299    SC  Importados/Indefinidos  ...            NaN        0.00
792607    RS  Importados/Indefinidos  ...            NaN        0.00
882774    SP  Importados/Indefinidos  ...            NaN        0.00

[19 rows x 9 columns]

Podemos ver que os códigos das cidades são nulos também. Vamos comparar com os registros dos estados, pois a documentação diz que o campo cidade como 'Importados/Indefinidos' pode se referir ao estado.

In [150]:
# listando os registros que têm a população nula
dados[dados['place_type']=='state']

state city place_type  ...  estimated_population  city_ibge_code  death_rate
0         AP  NaN      state  ...             861773.00              16        0.01
3331      AC  NaN      state  ...             894470.00              12        0.02
7716      DF  NaN      state  ...            3055149.00              53        0.02
8445      CE  NaN      state  ...            9187103.00              23        0.03
46218     ES  NaN      state  ...            4064052.00              32        0.02
59062     BA  NaN      state  ...           14930634.00              29        0.02
133724    MS  NaN      state  ...            2809394.00              50        0.02
147011    MA  NaN      state  ...            7114598.00              21        0.02
189381    GO  NaN      state  ...            7113540.00              52        0.02
231934    MT  NaN      state  ...            3526220.00              51        0.03
243923    PA  NaN      state  ...            8690745.00              15        0.03
272824    PB  NaN      state  ...            4039277.00              25        0.02
313022    MG  NaN      state  ...           21292666.00              31        0.02
444690    AL  NaN      state  ...            3351543.00              27        0.02
464391    PE  NaN      state  ...            9616621.00              26        0.05
501362    PI  NaN      state  ...            3281480.00              22        0.02
540872    PR  NaN      state  ...           11516840.00              41        0.02
603242    RO  NaN      state  ...            1796460.00              11        0.02
612296    AM  NaN      state  ...            4207714.00              13        0.03
624535    RJ  NaN      state  ...           17366189.00              33        0.07
643280    RN  NaN      state  ...            3534165.00              24        0.03
667764    RR  NaN      state  ...             631181.00              14        0.01
670884    SE  NaN      state  ...            2318822.00              28        0.03
684762    SC  NaN      state  ...            7252502.00              42        0.01
737946    TO  NaN      state  ...            1590248.00              17        0.01
758226    RS  NaN      state  ...           11422973.00              43        0.02
840955    SP  NaN      state  ...           46289333.00              35        0.03

[27 rows x 9 columns]

Vamos ver quantos registros existem com coluna com valor 'Importados/Indefinidos'.

In [151]:
# verificar quantos registros com valor city igual a 'Importados/Indefinidos'
dados[dados.city == 'Importados/Indefinidos'].shape[0]

19

Chegamos a conclusão que podemos apagar os registros com cidade igual a 'Importados/Indefinidos', pois eles não têm quantidades de casos maiores do que os dados dos estados, portanto, não são dados mais novos. Eles também não são relevantes como dados dos municípios, pois não têm código do município nem nome. Vamos apagar.

In [152]:
# apagando os registros
dados = dados.drop(dados[dados.city == 'Importados/Indefinidos'].index)
# exibir campos que têm valores nulos
dados.isnull().sum()

state                    0
city                    27
place_type               0
confirmed                0
deaths                   0
is_last                  0
estimated_population     0
city_ibge_code           0
death_rate               0
dtype: int64

Vamos alterar o nome dos campos **'state'** para **'sigla'** e o campo **'city'** para **'municipio'** para ficar coerente com os outros dataframes

In [153]:
# alterando os nomes das colunas
dados.rename(columns={'state': 'sigla', 'city': 'municipio' }, inplace = True)
# exibindo os três primeiros registros 
dados.head(3)

sigla municipio place_type  ...  estimated_population  city_ibge_code  death_rate
0      AP       NaN      state  ...             861773.00              16        0.01
232    AP     Amapá       city  ...               9187.00         1600105        0.01
419    AP  Calçoene       city  ...              11306.00         1600204        0.00

[3 rows x 9 columns]

Vamos dividir os dados em dados dos municipios e dados dos estados.

In [154]:
# criando o dataframe com dados dos estados
dados_estados = dados[dados['place_type'] == 'state']
dados_estados.head(3)

sigla municipio  ... city_ibge_code  death_rate
0       AP       NaN  ...             16        0.01
3331    AC       NaN  ...             12        0.02
7716    DF       NaN  ...             53        0.02

[3 rows x 9 columns]

In [155]:
# verificando a quantidade de registros
dados_estados.shape

(27, 9)

In [156]:
# criando o dataframe com dados dos municipios
dados_municipios = dados[dados['place_type'] == 'city']
dados_municipios.head(3)

sigla municipio place_type  ...  estimated_population  city_ibge_code  death_rate
232    AP     Amapá       city  ...               9187.00         1600105        0.01
419    AP  Calçoene       city  ...              11306.00         1600204        0.00
601    AP    Cutias       city  ...               6101.00         1600212        0.00

[3 rows x 9 columns]

In [157]:
# verificando a quantidade de registros
dados_municipios.shape

(5568, 9)

####Agora vamos pegar os dados dos outros arquivos e juntar a este.


#####**1) Área**
Vamos visualizar as primeiras linhas dos dois dataframes, o dos dados da COVID-19 dos estados e o de suas áreas

In [158]:
# visualizando dados da covid dos estados
dados_estados.head(3)

sigla municipio  ... city_ibge_code  death_rate
0       AP       NaN  ...             16        0.01
3331    AC       NaN  ...             12        0.02
7716    DF       NaN  ...             53        0.02

[3 rows x 9 columns]

In [159]:
# visualizando dados das áreas dos estados
estados_areas.head()

cod_uf    estado sigla       area
0     11  Rondônia    RO  237765.24
1     12      Acre    AC  164123.96
2     13  Amazonas    AM 1559167.89
3     14   Roraima    RR  223644.53
4     15      Pará    PA 1245870.80

Vamos juntar os dois pela sigla do estado

In [160]:
# juntando as colunas pela sigla dos estados
dados_estados = pd.merge(dados_estados, estados_areas, how='left', on='sigla')
dados_estados

sigla municipio place_type  ...  cod_uf               estado       area
0     AP       NaN      state  ...      16                Amapá  142470.76
1     AC       NaN      state  ...      12                 Acre  164123.96
2     DF       NaN      state  ...      53     Distrito Federal    5760.78
3     CE       NaN      state  ...      23                Ceará  148894.44
4     ES       NaN      state  ...      32       Espirito Santo   46074.45
5     BA       NaN      state  ...      29                Bahia  564760.43
6     MS       NaN      state  ...      50   Mato Grosso do Sul  357145.53
7     MA       NaN      state  ...      21             Maranhão  329642.18
8     GO       NaN      state  ...      52                Goiás  340203.33
9     MT       NaN      state  ...      51          Mato Grosso  903207.02
10    PA       NaN      state  ...      15                 Pará 1245870.80
11    PB       NaN      state  ...      25              Paraíba   56467.24
12    MG       NaN      state  ...      31         Minas Gerais  586521.12
13    AL       NaN      state  ...      27              Alagoas   27843.29
14    PE       NaN      state  ...      26           Pernambuco   98067.88
15    PI       NaN      state  ...      22                Piauí  251756.52
16    PR       NaN      state  ...      41               Paraná  199298.98
17    RO       NaN      state  ...      11             Rondônia  237765.24
18    AM       NaN      state  ...      13             Amazonas 1559167.89
19    RJ       NaN      state  ...      33       Rio de Janeiro   43750.43
20    RN       NaN      state  ...      24  Rio Grande do Norte   52809.60
21    RR       NaN      state  ...      14              Roraima  223644.53
22    SE       NaN      state  ...      28              Sergipe   21925.42
23    SC       NaN      state  ...      42       Santa Catarina   95730.68
24    TO       NaN      state  ...      17            Tocantins  277466.76
25    RS       NaN      state  ...      43    Rio Grande do Sul  281707.16
26    SP       NaN      state  ...      35            São Paulo  248219.48

[27 rows x 12 columns]

Vamos fazer o mesmo com os dados da COVID-19 dos municípios e suas áreas

In [161]:
# visualizando dados da covid dos municípios
dados_municipios.head(3)

sigla municipio place_type  ...  estimated_population  city_ibge_code  death_rate
232    AP     Amapá       city  ...               9187.00         1600105        0.01
419    AP  Calçoene       city  ...              11306.00         1600204        0.00
601    AP    Cutias       city  ...               6101.00         1600212        0.00

[3 rows x 9 columns]

In [162]:
# verificando a quantidade de registros
dados_municipios.shape

(5568, 9)

In [163]:
# visualizando dados das áreas dos municípios
municipios_areas.head()

cod_uf    estado sigla cod_munic              municipio    area
0     11  Rondônia    RO   1100015  ALTA FLORESTA D'OESTE 7067.02
1     11  Rondônia    RO   1100023              ARIQUEMES 4426.57
2     11  Rondônia    RO   1100031                 CABIXI 1314.35
3     11  Rondônia    RO   1100049                 CACOAL 3792.89
4     11  Rondônia    RO   1100056             CEREJEIRAS 2783.30

In [164]:
# verificando a quantidade de registros
municipios_areas.shape

(5572, 6)

Nos dados da COVID-19, vamos alterar o nome do campo **'city_ibge_code'** para **'cod_munic'** para ambos arquivos os campos ficarem com mesmo nome. 

In [165]:
# alterando os nomes das colunas
dados_municipios.rename(columns={'city_ibge_code': 'cod_munic'}, inplace = True)
# exibindo os três primeiros registros 
dados_municipios.head(3)

sigla municipio place_type  ...  estimated_population  cod_munic  death_rate
232    AP     Amapá       city  ...               9187.00    1600105        0.01
419    AP  Calçoene       city  ...              11306.00    1600204        0.00
601    AP    Cutias       city  ...               6101.00    1600212        0.00

[3 rows x 9 columns]

Vamos juntá-los pelos códigos dos municípios. A junção vai usar como referência os dados da COVID-19, ou seja, mesmo sabendo que o dataframe de áreas tem mais registros, que são únicos para os municípios, somente ficaram os registros dos municípios que constam nos dados da COVID.

In [166]:
# juntando as colunas pelos códigos dos municípios
dados_municipios = pd.merge(dados_municipios, municipios_areas, how='left', on='cod_munic')
dados_municipios

sigla_x         municipio_x  ...         municipio_y     area
0         AP               Amapá  ...               AMAPÁ  8454.85
1         AP            Calçoene  ...            CALÇOENE 14117.30
2         AP              Cutias  ...              CUTIAS  2179.11
3         AP      Ferreira Gomes  ...      FERREIRA GOMES  4973.85
4         AP             Itaubal  ...             ITAUBAL  1622.87
...      ...                 ...  ...                 ...      ...
5563      SP  Águas de São Pedro  ...  ÁGUAS DE SÃO PEDRO     3.61
5564      SP    Álvares Florence  ...    ÁLVARES FLORENCE   362.41
5565      SP     Álvares Machado  ...     ÁLVARES MACHADO   347.65
5566      SP  Álvaro de Carvalho  ...  ÁLVARO DE CARVALHO   153.66
5567      SP                Óleo  ...                ÓLEO   198.94

[5568 rows x 14 columns]

#####**2) IDHM** 
Vamos visualizar as primeiras linhas dos dois dataframes, o dos dados da COVID-19 dos estados e o de seus IDHM

In [167]:
# visualizando dados  da covid dos estados
dados_estados.head(3)

sigla municipio place_type  ...  cod_uf            estado      area
0    AP       NaN      state  ...      16             Amapá 142470.76
1    AC       NaN      state  ...      12              Acre 164123.96
2    DF       NaN      state  ...      53  Distrito Federal   5760.78

[3 rows x 12 columns]

In [168]:
# visualizar a quantidade de registros
dados_estados.shape

(27, 12)

In [169]:
# visualizando dados de IDHM dos estados
idhm_estados.head(3)

estado  idhm
0  Distrito Federal  0.85
1         São Paulo  0.83
2    Santa Catarina  0.81

In [170]:
# visualizar a quantidade de registros
idhm_estados.shape

(27, 2)

Vamos juntar os dois pelos nomes dos estados.

In [171]:
# juntando os dataframes
dados_estados = pd.merge(dados_estados, idhm_estados, how='left', on='estado')
# visualizando o dataframe modificado
dados_estados

sigla municipio place_type  ...               estado       area  idhm
0     AP       NaN      state  ...                Amapá  142470.76  0.74
1     AC       NaN      state  ...                 Acre  164123.96  0.72
2     DF       NaN      state  ...     Distrito Federal    5760.78  0.85
3     CE       NaN      state  ...                Ceará  148894.44  0.73
4     ES       NaN      state  ...       Espirito Santo   46074.45   nan
5     BA       NaN      state  ...                Bahia  564760.43  0.71
6     MS       NaN      state  ...   Mato Grosso do Sul  357145.53  0.77
7     MA       NaN      state  ...             Maranhão  329642.18  0.69
8     GO       NaN      state  ...                Goiás  340203.33  0.77
9     MT       NaN      state  ...          Mato Grosso  903207.02  0.77
10    PA       NaN      state  ...                 Pará 1245870.80  0.70
11    PB       NaN      state  ...              Paraíba   56467.24  0.72
12    MG       NaN      state  ...         Minas Gerais  586521.12  0.79
13    AL       NaN      state  ...              Alagoas   27843.29  0.68
14    PE       NaN      state  ...           Pernambuco   98067.88  0.73
15    PI       NaN      state  ...                Piauí  251756.52  0.70
16    PR       NaN      state  ...               Paraná  199298.98  0.79
17    RO       NaN      state  ...             Rondônia  237765.24  0.72
18    AM       NaN      state  ...             Amazonas 1559167.89  0.73
19    RJ       NaN      state  ...       Rio de Janeiro   43750.43  0.80
20    RN       NaN      state  ...  Rio Grande do Norte   52809.60  0.73
21    RR       NaN      state  ...              Roraima  223644.53  0.75
22    SE       NaN      state  ...              Sergipe   21925.42  0.70
23    SC       NaN      state  ...       Santa Catarina   95730.68  0.81
24    TO       NaN      state  ...            Tocantins  277466.76  0.74
25    RS       NaN      state  ...    Rio Grande do Sul  281707.16  0.79
26    SP       NaN      state  ...            São Paulo  248219.48  0.83

[27 rows x 13 columns]

O Estado do Espírito Santo está com IDHM nulo. Pode ter uma pequena diferença de grafia no nome do estado nos dois dataframes que provocou isso. Vamos verificar no dataframe de IDHM e inserir o valor.

In [172]:
# visualizando o dataframe de idhm
idhm_estados

estado  idhm
0      Distrito Federal  0.85
1             São Paulo  0.83
2        Santa Catarina  0.81
3        Rio de Janeiro  0.80
4                Paraná  0.79
5          Minas Gerais  0.79
6     Rio Grande do Sul  0.79
7           Mato Grosso  0.77
8        Espírito Santo  0.77
9                 Goiás  0.77
10   Mato Grosso do Sul  0.77
11              Roraima  0.75
12            Tocantins  0.74
13                Amapá  0.74
14                Ceará  0.73
15             Amazonas  0.73
16  Rio Grande do Norte  0.73
17           Pernambuco  0.73
18             Rondônia  0.72
19              Paraíba  0.72
20                 Acre  0.72
21                Bahia  0.71
22              Sergipe  0.70
23                 Pará  0.70
24                Piauí  0.70
25             Maranhão  0.69
26              Alagoas  0.68

In [173]:
# atribuindo o valor ao registro específico
dados_estados.loc[4, 'idhm'] = idhm_estados.loc[8, 'idhm']
# visualizando o registro nos dados da covid
dados_estados.loc[4, 'idhm']

0.772

Vamos fazer o mesmo para os municípios.

In [174]:
# visualizando dados da covid dos minicípios
dados_municipios.head(3)

sigla_x municipio_x place_type  confirmed  ...  estado  sigla_y  municipio_y     area
0      AP       Amapá       city        787  ...   Amapá       AP        AMAPÁ  8454.85
1      AP    Calçoene       city       1202  ...   Amapá       AP     CALÇOENE 14117.30
2      AP      Cutias       city        620  ...   Amapá       AP       CUTIAS  2179.11

[3 rows x 14 columns]

In [175]:
# visualizando a qantidade de registros
dados_municipios.shape

(5568, 14)

In [176]:
# visualizando dados de IDHM dos minicípios
idhm_municipios.head(3)

municipio  idhm  uf
0  SÃO CAETANO DO SUL  0.86  SP
1  ÁGUAS DE SÃO PEDRO  0.85  SP
2       FLORIANÓPOLIS  0.85  SC

In [177]:
# visualizando a qantidade de registros
idhm_municipios.shape

(5565, 3)

Vamos renomear a coluna do nome do municipio e a sigla do estado no dataframe para ficar igual ao dataframe de IDHM e vamos juntar os dois, considerando esses dois campos. Vamos também apagar uma das colunas que contém as siglas dos estados.

É necessário também renomear o campo uf do dataframe de IDHM para sigla.

In [178]:
# excluindo uma das colunas de siglas dos estados
del dados_municipios['sigla_x']
# alterando os nomes das colunas
dados_municipios.rename(columns={'municipio_y': 'municipio', 'sigla_y': 'sigla'}, inplace = True)
# exibindo os três primeiros registros 
dados_municipios.head(3)

municipio_x place_type  confirmed  deaths  ...  estado  sigla municipio     area
0       Amapá       city        787       4  ...   Amapá     AP     AMAPÁ  8454.85
1    Calçoene       city       1202       5  ...   Amapá     AP  CALÇOENE 14117.30
2      Cutias       city        620       2  ...   Amapá     AP    CUTIAS  2179.11

[3 rows x 13 columns]

In [179]:
# renomeando o campo uf
idhm_municipios.rename(columns={'uf': 'sigla'}, inplace = True)
# visualizando o dataframe
idhm_municipios.head(3)

municipio  idhm sigla
0  SÃO CAETANO DO SUL  0.86    SP
1  ÁGUAS DE SÃO PEDRO  0.85    SP
2       FLORIANÓPOLIS  0.85    SC

In [180]:
# juntando os dataframes
dados_municipios = pd.merge(dados_municipios, idhm_municipios, how='left', on=['municipio', 'sigla'])
# visualizando o dataframe modificado
dados_municipios

municipio_x place_type  ...     area  idhm
0                  Amapá       city  ...  8454.85  0.64
1               Calçoene       city  ... 14117.30  0.64
2                 Cutias       city  ...  2179.11  0.63
3         Ferreira Gomes       city  ...  4973.85  0.66
4                Itaubal       city  ...  1622.87  0.58
...                  ...        ...  ...      ...   ...
5563  Águas de São Pedro       city  ...     3.61  0.85
5564    Álvares Florence       city  ...   362.41  0.73
5565     Álvares Machado       city  ...   347.65  0.76
5566  Álvaro de Carvalho       city  ...   153.66  0.69
5567                Óleo       city  ...   198.94  0.73

[5568 rows x 14 columns]

Como o dataframe de IDHM possuia 5.565 registros e o de dados da COVID-19 5.568 registros, deve haver valores nulos. Vamos verificar.

In [181]:
# visualizando a quantidade de registros nulos e os cinco primeiros registros
print(f'Registros nulos: {dados_municipios[dados_municipios.idhm.isnull()].shape[0]}')
dados_municipios[dados_municipios.idhm.isnull()].head()

Registros nulos: 40


municipio_x place_type  confirmed  ...        municipio    area  idhm
93             Ereré       city        364  ...            ERERÉ  362.91   nan
125          Itapajé       city       1177  ...          ITAPAJÉ  432.19   nan
230  Atilio Vivacqua       city        352  ...  ATÍLIO VIVACQUA  232.87   nan
327           Araçás       city        267  ...           ARAÇÁS  474.58   nan
503             Iuiu       city         73  ...             IUIU 1525.14   nan

[5 rows x 14 columns]

Provavelmente os dados são oriundos dos nomes diferentes na grafia nos dois arquivos que geraram os dataframes. Infelizmente vamos deixar como está, mas a solução seria buscar essa difereça e coriigir ou buscar um arquivo dos dados que possua o código IBGE do município. Isso fica como trabalho futuro.

#####**3) Longitude e Latitude**
Vamos visualizar os dados de COVID-19 dos municípios e de coordenadas geográficas dos municípios.

In [182]:
# visualizando os dados de covid dos municípios
dados_municipios.head(3)

municipio_x place_type  confirmed  deaths  ...  sigla  municipio     area  idhm
0       Amapá       city        787       4  ...     AP      AMAPÁ  8454.85  0.64
1    Calçoene       city       1202       5  ...     AP   CALÇOENE 14117.30  0.64
2      Cutias       city        620       2  ...     AP     CUTIAS  2179.11  0.63

[3 rows x 14 columns]

In [183]:
# visualizando a quantidade de registros
dados_municipios.shape

(5568, 14)

In [184]:
# visualizando os dados das coordenadas geográficas
latlong_municipios.head()

cod_municipio              municipio  longitude  latitude
0       1100015  ALTA FLORESTA D'OESTE     -62.00    -11.94
1       1100023              ARIQUEMES     -63.03     -9.91
2       1100031                 CABIXI     -60.54    -13.50
3       1100049                 CACOAL     -61.44    -11.43
4       1100056             CEREJEIRAS     -60.82    -13.20

In [185]:
# visualizando a quantidade de registros
latlong_municipios.shape

(5570, 4)

Vamos renomear a coluna do município para ficarem iguais a dos dados da COVID-19. Vamos, em seguida juntar os dataframes pelos códigos dos municípios. Podemos também apagar a coluna municipio dos dados de COVID-19

In [186]:
# apagando a coluna municipio do dataframe de covid
del dados_municipios['municipio']
# alterando o nome da coluna do código IBGE do município
latlong_municipios.rename(columns={'cod_municipio': 'cod_munic'}, inplace=True)
# vamos juntar os dois dataframes
pd.merge(dados_municipios, latlong_municipios, how='left', on='cod_munic')

municipio_x place_type  ...  longitude  latitude
0                  Amapá       city  ...     -50.79      2.05
1               Calçoene       city  ...     -50.94      2.50
2                 Cutias       city  ...     -50.80      0.97
3         Ferreira Gomes       city  ...     -51.19      0.85
4                Itaubal       city  ...     -50.70      0.60
...                  ...        ...  ...        ...       ...
5563  Águas de São Pedro       city  ...     -47.88    -22.60
5564    Álvares Florence       city  ...     -49.91    -20.32
5565     Álvares Machado       city  ...     -51.47    -22.08
5566  Álvaro de Carvalho       city  ...     -49.72    -22.08
5567                Óleo       city  ...     -49.34    -22.94

[5568 rows x 16 columns]

#####**4) Calculando a densidade demográfica**

A densidade demográfica é a quantidade da população dividida pela área. Vamos ter que criar uma coluna para receber esse valor, fazer o cáculo e atribuir o valor a ela. Precisaremos analisar se há valores nulos em área e população. Isso será feito para os dados dos estados e para os dados dos municípios.

In [187]:
# criando a coluna densidade nos dois dataframes (estados e municípios)
dados_estados['densidade'] =  0.0
dados_municipios['densidade'] = 0.0

In [188]:
# verificando se há valores nulos nos estados
dados_estados.isnull().sum()

sigla                    0
municipio               27
place_type               0
confirmed                0
deaths                   0
is_last                  0
estimated_population     0
city_ibge_code           0
death_rate               0
cod_uf                   0
estado                   0
area                     0
idhm                     0
densidade                0
dtype: int64

In [189]:
# verificando se há valores nulos nos estados
dados_municipios.isnull().sum()

municipio_x              0
place_type               0
confirmed                0
deaths                   0
is_last                  0
estimated_population     0
cod_munic                0
death_rate               0
cod_uf                   0
estado                   0
sigla                    0
area                     0
idhm                    40
densidade                0
dtype: int64

Não tendo valores nulos de área e de populacao, vamos calcular, em ambos dataframes, as densidades demográficas.

In [190]:
# calculando a densidade demografica dos estados
dados_estados.densidade = dados_estados.estimated_population/dados_estados.area
dados_estados.head(3)

sigla municipio place_type  ...      area  idhm  densidade
0    AP       NaN      state  ... 142470.76  0.74       6.05
1    AC       NaN      state  ... 164123.96  0.72       5.45
2    DF       NaN      state  ...   5760.78  0.85     530.34

[3 rows x 14 columns]

In [191]:
# calculando a densidade demografica dos municípios
dados_municipios.densidade = dados_municipios.estimated_population/dados_municipios.area
dados_municipios.head(3)

municipio_x place_type  confirmed  deaths  ...  sigla     area idhm  densidade
0       Amapá       city        787       4  ...     AP  8454.85 0.64       1.09
1    Calçoene       city       1202       5  ...     AP 14117.30 0.64       0.80
2      Cutias       city        620       2  ...     AP  2179.11 0.63       2.80

[3 rows x 14 columns]

#####**5) Verificação das colunas**

Vamos verificar as colunas dos dois dataframes e excluir as desnecessárias

In [192]:
# visualizando os dados dos estados
dados_estados.head(3)

sigla municipio place_type  ...      area  idhm  densidade
0    AP       NaN      state  ... 142470.76  0.74       6.05
1    AC       NaN      state  ... 164123.96  0.72       5.45
2    DF       NaN      state  ...   5760.78  0.85     530.34

[3 rows x 14 columns]

Vamos apagar as colunas do nome do município (**municipio**), do tipo de lugar (**place_type**), de indicação de último lançamento (**is_last**) e do código IBGE da cidade (**city_ibge_code**), pois são desnecessárias a partir de agora.

In [193]:
# apagando as colunas municipio, place_type, is_last e city_ibge_code
dados_estados.drop(['municipio', 'place_type', 'is_last', 'city_ibge_code'], axis=1, inplace=True)

# visualizando os dados dos estados
dados_estados.head(3)

sigla  confirmed  deaths  ...      area  idhm densidade
0    AP      53790     770  ... 142470.76  0.74      6.05
1    AC      32113     705  ... 164123.96  0.72      5.45
2    DF     217793    3788  ...   5760.78  0.85    530.34

[3 rows x 10 columns]

In [194]:
# visualizando os dados dos municípios
dados_municipios.head(3)

municipio_x place_type  confirmed  deaths  ...  sigla     area idhm  densidade
0       Amapá       city        787       4  ...     AP  8454.85 0.64       1.09
1    Calçoene       city       1202       5  ...     AP 14117.30 0.64       0.80
2      Cutias       city        620       2  ...     AP  2179.11 0.63       2.80

[3 rows x 14 columns]

Vamos apagar as colunas do tipo de lugar (**place_type**) e de indicação de último lançamento (**is_last**), pois são desnecessárias a partir de agora. Vamos também renomear a coluna do nome do município.

In [195]:
# apagando as colunas place_type e is_last
dados_municipios.drop(['place_type', 'is_last'], axis=1, inplace=True)
# renomeando a coluna com nomes dos municípios
dados_municipios.rename(columns={'municipio_x': 'municipio'}, inplace=True)
# visualizando os dados dos estados
dados_municipios.head(3)

municipio  confirmed  deaths  ...     area idhm  densidade
0     Amapá        787       4  ...  8454.85 0.64       1.09
1  Calçoene       1202       5  ... 14117.30 0.64       0.80
2    Cutias        620       2  ...  2179.11 0.63       2.80

[3 rows x 12 columns]

### **Obtendo as Informações a partir dos Dados**

#### **1) Cidades com maiores e menores casos confirmados e mortes**

In [196]:
# exibindo as dez cidades com menos casos confirmados 
colunas_selecionadas = ['municipio', 'sigla', 'confirmed']
print('As dez cidades com menos casos confirmados:')

dados_municipios[colunas_selecionadas].sort_values('confirmed',ascending=True).head(10)

As dez cidades com menos casos confirmados:


municipio sigla  confirmed
3314             Laranjal    PR          0
2095            Ibertioga    MG          1
2483  Serra Azul de Minas    MG          1
4521         Cerro Branco    RS          1
4894            Ubiretama    RS          1
1800             Araponga    MG          1
1322              Luciara    MT          1
2601            Veredinha    MG          1
1975        Coronel Murta    MG          1
2557  São Tomé das Letras    MG          1

Vamos verificar quantas cidades possuem um único caso confirmado.

In [197]:
# exibindo a quantidade de cidades que tem uma ou nenhuma
print(f'{dados_municipios[dados_municipios.confirmed == 1].shape[0]} cidades de 5.570 tiveram um caso confirmado.')

13 cidades de 5.570 tiveram um caso confirmado.


Agora vamos listar as 10 cidades com mais casos confirmados.

In [198]:
# exibindo as dez cidades com mais casos confirmados 
print('As dez cidades com mais casos confirmados:')

dados_municipios[colunas_selecionadas].sort_values('confirmed',ascending=False).head(10)

As dez cidades com mais casos confirmados:


municipio sigla  confirmed
5495       São Paulo    SP     325408
38          Brasília    DF     191012
3710  Rio de Janeiro    RJ     118433
634         Salvador    BA      92988
1108         Goiânia    GO      67955
3618          Manaus    AM      63557
97         Fortaleza    CE      57675
1829  Belo Horizonte    MG      49038
1419           Belém    PA      46129
4747    Porto Alegre    RS      43472

Vamos exibir as 10 cidades com menos e mais mortes por COVID-19

In [199]:
# exibindo as dez cidades com menos mortes 
colunas_selecionadas = ['municipio', 'sigla', 'deaths']
print('As dez cidades com menos mortes:')

dados_municipios[colunas_selecionadas].sort_values('deaths',ascending=True).head(10)

As dez cidades com menos mortes:


municipio sigla  deaths
5567                   Óleo    SP       0
1366             Rio Branco    MT       0
1365          Ribeirãozinho    MT       0
1363     Reserva do Cabaçal    MT       0
4149            Matos Costa    SC       0
1356  Porto Alegre do Norte    MT       0
1353     Pontal do Araguaia    MT       0
1351      Planalto da Serra    MT       0
1367            Rondolândia    MT       0
4153                 Mondaí    SC       0

Vamos levantar quantas cidades não tiveram mortes.

In [200]:
# quantidade de registros com valores de mortes iguais a zero
print(f'{dados_municipios[dados_municipios.deaths == 0].shape[0]} de um total de 5.570 cidades não tiveram mortes registradas.')

831 de um total de 5.570 cidades não tiveram mortes registradas.


Vamos listar as cidades com mais mortes pela COVID-19

In [201]:
# exibindo as dez cidades com mais mortes 
colunas_selecionadas = ['municipio', 'sigla', 'deaths']
print('As dez cidades com mais mortes:')

dados_municipios[colunas_selecionadas].sort_values('deaths',ascending=False).head(10)

As dez cidades com mais mortes:


municipio sigla  deaths
5495       São Paulo    SP   13767
3710  Rio de Janeiro    RJ   12064
97         Fortaleza    CE    3930
38          Brasília    DF    3481
634         Salvador    BA    2932
3618          Manaus    AM    2902
2851          Recife    PE    2487
1419           Belém    PA    2267
1108         Goiânia    GO    1734
5127       Guarulhos    SP    1549

Gráficos

#### **2) Quantidade de casos confirmados e mortes nos estados, em ordem crescente**

In [203]:
# exibindo a lista dos estados em ordem crescente de casos confirmados
colunas_selecionadas = ['estado', 'confirmed']
print('Casos confirmados em cada Estado:')

dados_estados[colunas_selecionadas].sort_values('confirmed', ascending=True)

Casos confirmados em cada Estado:


estado  confirmed
1                  Acre      32113
0                 Amapá      53790
21              Roraima      59586
17             Rondônia      73990
24            Tocantins      77306
20  Rio Grande do Norte      83309
22              Sergipe      85643
6    Mato Grosso do Sul      85990
13              Alagoas      92040
15                Piauí     118939
11              Paraíba     136637
9           Mato Grosso     148623
4        Espirito Santo     165688
18             Amazonas     167486
14           Pernambuco     168093
7              Maranhão     188677
2      Distrito Federal     217793
16               Paraná     225624
10                 Pará     260058
8                 Goiás     265131
25    Rio Grande do Sul     267750
3                 Ceará     279989
23       Santa Catarina     283252
19       Rio de Janeiro     320598
5                 Bahia     367669
12         Minas Gerais     374651
26            São Paulo    1150872

In [204]:
# exibindo a lista dos estados em ordem crescente de mortes
colunas_selecionadas = ['estado', 'deaths']
print('Mortes em cada Estado:')

dados_estados[colunas_selecionadas].sort_values('deaths', ascending=True)

Mortes em cada Estado:


estado  deaths
21              Roraima     701
1                  Acre     705
0                 Amapá     770
24            Tocantins    1120
17             Rondônia    1486
6    Mato Grosso do Sul    1645
22              Sergipe    2247
13              Alagoas    2282
15                Piauí    2499
20  Rio Grande do Norte    2620
11              Paraíba    3178
23       Santa Catarina    3268
2      Distrito Federal    3788
9           Mato Grosso    3943
4        Espirito Santo    3969
7              Maranhão    4146
18             Amazonas    4655
16               Paraná    5563
8                 Goiás    5900
25    Rio Grande do Sul    6034
10                 Pará    6806
5                 Bahia    7859
14           Pernambuco    8773
12         Minas Gerais    9204
3                 Ceará    9418
19       Rio de Janeiro   20970
26            São Paulo   39907

Gráficos

#### **3) Cidades com maiores e menores casos confirmados e mortes por 1000 habitantes**
Vamos criar as colunas de casos confirmados e mortes por 1000 habitantes e atribuir o valor calculado para elas, exibindo as informações.

In [495]:
# criando a coluna de casos confirmados, atribuindo o valor calculado e exibindo as informações
dados_municipios['confirmed_per_1000_inhabitants'] = 1000*dados_municipios['confirmed']/dados_municipios['estimated_population']

In [496]:
# considerado que valores igual ou maior que 0.5 é igual a 1 caso
qtde = dados_municipios.query('confirmed_per_1000_inhabitants < 0.5').shape[0]
# total dos casos
total = dados_municipios.shape[0]
print(f'{qtde} de {total} cidades possuem menos de 1 caso confirmado a cada 1000 habitantes. São elas:')
# exibindo a lista de cidades e sigla dos estados a que pertencem
colunas_selecionadas = ['municipio', 'sigla']
dados_municipios.query('confirmed_per_1000_inhabitants < 0.5')[colunas_selecionadas]

21 de 5568 cidades possuem menos de 1 caso confirmado a cada 1000 habitantes. São elas:


municipio sigla
354            Brotas de Macaúbas    BA
717                 Érico Cardoso    BA
1322                      Luciara    MT
1800                     Araponga    MG
1811                   Aricanduva    MG
1892                        Canaã    MG
1938             Chapada do Norte    MG
1975                Coronel Murta    MG
2057             Francisco Badaró    MG
2095                    Ibertioga    MG
2165            Jenipapo de Minas    MG
2200                Leme do Prado    MG
2411                 Rio Vermelho    MG
2442      Santa Rita de Ibitipoca    MG
2477  Senador Modestino Gonçalves    MG
2483          Serra Azul de Minas    MG
2557          São Tomé das Letras    MG
2601                    Veredinha    MG
3314                     Laranjal    PR
4521                 Cerro Branco    RS
4737                Pinhal Grande    RS

In [505]:
# exibindo as dez cidades com mais casos confirmados por 1000 habitantes
colunas_selecionadas = ['municipio', 'sigla', 'confirmed_per_1000_inhabitants']
print('As dez cidades com maiores números de casos confirmados a cada 1000 habitantes foram:')

dados_municipios[colunas_selecionadas].sort_values('confirmed_per_1000_inhabitants',ascending=False).head(10)

As dez cidades com maiores números de casos confirmados a cada 1000 habitantes foram:


municipio sigla  confirmed_per_1000_inhabitants
1459             Jacareacanga    PA                          223.98
3612                   Japurá    AM                          215.46
15            Vitória do Jari    AP                          179.03
9     Pedra Branca do Amapari    AP                          164.41
1200                Porteirão    GO                          155.18
4786     Santa Cecília do Sul    RS                          141.98
4694               Nova Araçá    RS                          132.20
1488              Parauapebas    PA                          131.05
865    Feira Nova do Maranhão    MA                          130.83
1801                  Araporã    MG                          130.57

Exibindo três gráficos: com os dez municípios com maior número de casos confirmados, os dados de seus IDHM e os dados de densidade demográfica, 

Vamos agora levantar os dados de mortos dos municípios.

In [506]:
# criação da coluna de mortos por 1000 habitantes e atribuindo valor a ela
dados_municipios['deaths_per_1000_inhabitants'] = 1000*dados_municipios['deaths']/dados_municipios['estimated_population']
# quantidade de mortes por 1000 habitantes menor do que 1, considerando que quantidade maior ou igual a 0.5 é considerado 1
qtde = dados_municipios.query('deaths_per_1000_inhabitants < 0.5').shape[0]
total = dados_municipios.shape[0]
print(f'{qtde} de {total} cidades possuem menos de 1 morte a cada 1000 habitantes.')

3598 de 5568 cidades possuem menos de 1 morte a cada 1000 habitantes.


In [509]:
# exibindo as dez cidades com mais mortes por 1000 habitantes
colunas_selecionadas = ['municipio', 'sigla', 'deaths_per_1000_inhabitants']
print('As dez cidades com maiores números de mortes a cada 1000 habitantes foram:')
#for index, row in dados_municipios.sort_values('deaths_per_1000_inhabitants', ascending=True).tail(10).iterrows():
#    print(f'- {row.city}-{row.state}: aproximadamente {row.deaths_per_1000_inhabitants:.0f} ({row.deaths_per_1000_inhabitants:.1f});')
dados_municipios[colunas_selecionadas].sort_values('deaths_per_1000_inhabitants', ascending=False).head(10)

As dez cidades com maiores números de mortes a cada 1000 habitantes foram:


municipio sigla  deaths_per_1000_inhabitants
4527               Charrua    RS                         3.38
5056             Cosmorama    SP                         2.88
3564  Pimenteiras do Oeste    RO                         2.79
4679              Montauri    RS                         2.78
3129           Água Branca    PI                         2.63
5542                   Uru    SP                         2.60
1019              Aloândia    GO                         2.52
5009               Bálsamo    SP                         2.52
5108        Gastão Vidigal    SP                         2.47
5074          Dolcinópolis    SP                         2.37

Exibindo três gráficos: com os dez municípios com maior número de mortes, os dados de seus IDHM e os dados de densidade demográfica,

#### **4) Estados em ordem crescente de número de casos confirmados e mortes por 1000 habitantes**
Vamos criar as colunas de casos confirmados e mortes por 1000 habitantes e atribuir o valor calculado para elas, exibindo as informações.

In [517]:
# criando a coluna de casos confirmados, atribuindo o valor calculado e exibindo as informações
dados_estados['confirmed_per_1000_inhabitants'] = 1000*dados_estados['confirmed']/dados_estados['estimated_population']

In [520]:
# exibindo os 27 estados por ordem crescente de casos confirmados por 1000 habitantes
colunas_selecionadas = ['estado', 'confirmed_per_1000_inhabitants']
print('Casos confirmados por 1000 habitantes:')
dados_estados[colunas_selecionadas].sort_values('confirmed_per_1000_inhabitants',ascending=True)

Casos confirmados por 1000 habitantes:


estado  confirmed_per_1000_inhabitants
14           Pernambuco                           17.48
12         Minas Gerais                           17.60
19       Rio de Janeiro                           18.46
16               Paraná                           19.59
25    Rio Grande do Sul                           23.44
20  Rio Grande do Norte                           23.57
5                 Bahia                           24.63
26            São Paulo                           24.86
7              Maranhão                           26.52
13              Alagoas                           27.46
10                 Pará                           29.92
3                 Ceará                           30.48
6    Mato Grosso do Sul                           30.61
11              Paraíba                           33.83
1                  Acre                           35.90
15                Piauí                           36.25
22              Sergipe                           36.93
8                 Goiás                           37.27
23       Santa Catarina                           39.06
18             Amazonas                           39.80
4        Espirito Santo                           40.77
17             Rondônia                           41.19
9           Mato Grosso                           42.15
24            Tocantins                           48.61
0                 Amapá                           62.42
2      Distrito Federal                           71.29
21              Roraima                           94.40

In [519]:
# criando a coluna de mortes, atribuindo o valor calculado e exibindo as informações
dados_estados['deaths_per_1000_inhabitants'] = 1000*dados_estados['deaths']/dados_estados['estimated_population']

In [521]:
# exibindo os 27 estados por ordem crescente de mortes por 1000 habitantes
colunas_selecionadas = ['estado', 'deaths_per_1000_inhabitants']
print('Mortes por 1000 habitantes:')
dados_estados[colunas_selecionadas].sort_values('deaths_per_1000_inhabitants',ascending=True)

Mortes por 1000 habitantes:


estado  deaths_per_1000_inhabitants
12         Minas Gerais                         0.43
23       Santa Catarina                         0.45
16               Paraná                         0.48
5                 Bahia                         0.53
25    Rio Grande do Sul                         0.53
7              Maranhão                         0.58
6    Mato Grosso do Sul                         0.59
13              Alagoas                         0.68
24            Tocantins                         0.70
20  Rio Grande do Norte                         0.74
15                Piauí                         0.76
10                 Pará                         0.78
11              Paraíba                         0.79
1                  Acre                         0.79
17             Rondônia                         0.83
8                 Goiás                         0.83
26            São Paulo                         0.86
0                 Amapá                         0.89
14           Pernambuco                         0.91
22              Sergipe                         0.97
4        Espirito Santo                         0.98
3                 Ceará                         1.03
18             Amazonas                         1.11
21              Roraima                         1.11
9           Mato Grosso                         1.12
19       Rio de Janeiro                         1.21
2      Distrito Federal                         1.24

Gráficos casos confirmados, mortes, IDHM e densidade demográfica

#### **5) Há relação entre os casos confirmados e IDHM nos Estados**
Vamos montar gráficos que apoiem esta análise.

#### **6) Há relação entre os casos confirmados e densidade demográfica nos Estados**
Vamos montar gráficos que apoiem esta análise.

#### **7) Mapas com dados de casos confirmados e mortes por cidade**
Vamos montar gráficos que apoiem esta análise.

#### **8) Mapas com dados de casos confirmados e mortes por estado**
Vamos montar gráficos que apoiem esta análise.

## **Conclusão**

## **Referências**
* **[MinisteriodaSaude]** Sobre a Doença. Ministério da Saúde. Disponível em: <https://coronavirus.saude.gov.br/sobre-a-doenca>. Acesso em: 12 nov. 2020.
* **[WikipediaCOVID-19]** Pandemia de COVID-19. Wikipedia. Disponível em: <https://pt.wikipedia.org/wiki/Pandemia_de_COVID-19>. Acesso em: 13 nov. 2020.
* **[WikipediaSARS-CoV]** SARS-CoV. Wikipedia. Disponível em: <https://pt.wikipedia.org/wiki/SARS-CoV>. Acesso em: 13 nov. 2020.
* **[Sanar]** Linha do tempo do Coronavírus no Brasil. Sanar. Disponível em: <https://www.sanarmed.com/linha-do-tempo-do-coronavirus-no-brasil>. Acesso em: 13 nov. 2020.
* **[Lemos2020]** Vinícius Lemos. Coronavírus: por que primeira pessoa infectada no Brasil pode nunca ser descoberta. BBC News Brasil. Disponível em: <https://www.bbc.com/portuguese/brasil-52334034>. Acesso em: 13 nov. 2020.
* **[SUSAnalitico]** COVID-19 no Brasil. SUSAnalítico. Ministério da Saúde. Disponível em: <https://susanalitico.saude.gov.br/extensions/covid-19_html/covid-19_html.html>. Acesso em: 13 nov. 2020.
* **[OPASBrasil]** OMS afirma que COVID-19 é agora caracterizada como pandemia. Organização Pan-Americana de Saúde (OPAS Brasil). Disponível em: <https://www.paho.org/bra/index.php?option=com_content&view=article&id=6120:oms-afirma-que-covid-19-e-agora-caracterizada-como-pandemia&Itemid=812>. Acesso em: 13 nov. 2020.
* Bootcamp Data Science Aplicada. Bootcamps Alura. Disponível em: <https://www.alura.com.br/bootcamp/data-science-aplicada/matriculas-abertas>. Acesso em: 01 nov. 2020.
* GeoJSON vs JSON. Stackshare. Digital Ocean. Disponível em: <https://stackshare.io/stackups/geojson-vs-json>. Acesso em: 14 nov. 2020.
* JSON vs GJSON. Github do EarthCube CHORDS. Disponível em: <https://github.com/earthcubeprojects-chords/chords/wiki/JSON-vs-GeoJSON>. Acesso em: 14 nov. 2020.
* Lucas Oliveira. Descomplicando o GroupBy. Medium. Disponível em: <https://medium.com/@lucasoliveiras/descomplicando-o-groupby-f57ff2dcd6db>. Acesso em: 15 nov. 2020.
